### PR TITLE
✨ Bump CAPI to v1.8.0 pre alpha

### PR DESCRIPTION
**Pull request by kashifest**
This PR steers the Main branch towards CAPI main branch. Namely it bumps

- CAPI to v1.8.0 pre alpha (main branch commit that points to v1.7.0-rc.0.0.20240704092232-a7f5043dee2a)
- k8s.io/* to v0.30.2
- sigs.k8s.io/controller-runtime to v0.18.4

### CODE REVIEW

File: test/go.mod [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
-go 1.22
+go 1.22.3
 
 require (
-	github.com/docker/docker v26.1.4+incompatible
+	github.com/docker/docker v27.0.1-rc.1+incompatible
 	github.com/jinzhu/copier v0.4.0
 	github.com/metal3-io/baremetal-operator/apis v0.6.1
-	github.com/metal3-io/cluster-api-provider-metal3/api v0.0.0
+	github.com/metal3-io/cluster-api-provider-metal3/api v0.0.0-00010101000000-000000000000
```
Comment (tuminoid) [inline test/go.mod:9]: Where did this come from?

Comment (kashifest) [reply test/go.mod]: Strange! didn't notice it, I wonder what caused it

**Pull request merged**